modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderServiceImpl.java

```diff
@@ -86,6 +93,16 @@ public CommerceOrder addOrUpdateCommerceOrder(
 				externalReferenceCode, serviceContext.getCompanyId());
 
 		if (commerceOrder == null) {
+			CommerceAccount commerceAccount = _getCommerceAccount(
+				commerceAccountId);
+
+			if (commerceAccount.isBusinessAccount()) {
+				_portletResourcePermission.check(
+					getPermissionChecker(),
+					commerceAccount.getCommerceAccountGroupId(),
+					CommerceOrderActionKeys.ADD_COMMERCE_ORDER);
+			}
+
 			_portletResourcePermission.check(
```

Comment:
This is executed even if the account is a business type. It is not clear to me why we need to check 2 different permissions.